### ISSUE TITLE
USB class request callback stalls - potential problem

### ISSUE THREAD

**Issue by dougg3** [OpenNuvoton/M251BSP#1]
Hi! I found what I think is a bug in a lot of the USB samples. Many of the samples do this in the class request callback when there's an invalid class request:

```
USBD_SetStall(EP0);
USBD_SetStall(EP1);
```

For example, the USBD_VCOM_SerialEmulator sample does this. Isn't this wrong? Because it's going to stall whatever you've assigned to USB endpoint number 1 (the bulk in endpoint in this example), instead of EP1 as intended.

Some other samples (e.g. USBD_VCOM_And_Mass_Storage) only do: `USBD_SetStall(EP0)` which is safer, but only stalls one of the two control endpoints.

Shouldn't these class request callbacks do the same thing as the common USBD code (e.g. USBD_StandardRequest) when there's an invalid setup packet? Like this:

```
USBD_SET_EP_STALL(EP0);
USBD_SET_EP_STALL(EP1);
```

**Comment by chhsieh3** [comment]
Thank your feedback. We will fix them in the future release.

**Comment by dougg3** [comment]
Looks like the fix was just released with V3.02.005. Thanks!
